### PR TITLE
chore(ci) Add a GitHub actions workflow for our CI

### PR DESCRIPTION
**Pull request by joshgoebel**
Quick stab at a GHA CI workflow.  Travis CI has been ridiculously flaky lately.

### Checklist
- [x] Added markup tests, or they don't apply here because...
- [x] Updated the changelog at `CHANGES.md` (Not needed)
- [x] Added myself to `AUTHORS.txt`, under Contributors

### CODE REVIEW

**Comment by joshgoebel** [comment]
@allejo If we could do the browser builds as part of the matrix that would be even better... I think we could do that with a second key and `if`, but I'm not sure.  Though it already completes in just about a minute, which is great.  But faster is better with CI. :)

```
      matrix:
        node-version: [12.x, 14.x, 15.x]
        build_how: ["node", "browser", "browser -n"]
```

File: .github/workflows/node.js.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.node-version }}
+    # normal browser build
+    - run: npm install
+    - if: ${{ contains(matrix.build-how, 'browser') }}
```
Comment (allejo) [inline .github/workflows/node.js.yml:30]: just FYI, an `if` should be attached to a "step" and not by itself

Comment (allejo) [reply .github/workflows/node.js.yml:30]: also, `${{ }}` notation is automatically assumed in an `if` so you can just do contains

Comment (joshgoebel) [reply .github/workflows/node.js.yml:30]: > just FYI, an if should be attached to a "step" and not by itself 

Seems to work with multiple ifs inside `steps` doing different things.

**Comment by joshgoebel** [comment]
Screen Shot 2020-11-02 at 8 18 54 PM

Ok, yikes... that's a bit verbose... maybe I don't want such a complex matrix?  Any thoughts?

**Review by allejo — APPROVED** [review]
I'm okay with the output of the matrix, it's thorough with everything that's officially supported/tested

**Comment by allejo** [comment]
I love how the CI only takes a minute on GHA :heart:

**Comment by joshgoebel** [comment]
Is this not going to work on all the existing PRs though because the action doesn't exist on those branches?  That part confuses me.

**Comment by allejo** [comment]
That's the behavior that I've seen, yes. Current PRs will need to have `master` merged into them or rebased.

**Comment by joshgoebel** [comment]
> Current PRs will need to have master merged into them or rebased.

Ah, but typically that's just one click anyways. :-)  That'll work.

**Pull request merged**